core/config/project_settings.h

```diff
@@ -218,6 +222,9 @@ class ProjectSettings : public Object {
 	String get_scene_groups_cache_path() const;
 	void load_scene_groups_cache();
 
+	// Testing a version allows fast cached GET_GLOBAL macros.
+	uint32_t get_version() const { return _version; }
```

Comment:
Ah, I see. If the version is different, the cache needs to be updated. That's actually quite genius, but what about memory usage?

Reply:
What do you mean by 'memory usage'? You mean the cost of storing the version for each `GLOBAL_GET_CACHED` we use?

Reply:
Yeah!

Reply:
In the codebase we have around 725 uses of `GLOBAL_GET`. Each version costs 4 bytes. If we changed all of them with `GLOBAL_GET_CACHED`, it would cost 2.9kb for the version caches (and probably a bunch more for the value caches themselves).
It's not negligible, but we're wasting a whole lot more elsewhere (which I'm working on).